numpy/core/src/umath/ufunc_object.c

```diff
@@ -4387,25 +4403,36 @@ PyUFunc_GenericReduction(PyUFuncObject *ufunc, PyObject *args,
         return NULL;
     }
 
-    /* If an output parameter was provided, don't wrap it */
-    if (out != NULL) {
-        return (PyObject *)ret;
-    }
+    /* Wrap and return the output */
+    {
+        ufunc_full_args full_args;
+        PyObject *wrap;
+        PyObject *res;
 
-    if (Py_TYPE(op) != Py_TYPE(ret)) {
-        res = PyObject_CallMethod(op, "__array_wrap__", "O", ret);
-        if (res == NULL) {
-            PyErr_Clear();
+        /* Prepare an in/out arguments object */
+        full_args.in = PyTuple_Pack(1, op);
+        if (full_args.in == NULL) {
+            return NULL;
         }
-        else if (res == Py_None) {
-            Py_DECREF(res);
+        if (out != NULL) {
+            full_args.out = PyTuple_Pack(1, out);
```

Comment:
Always one output in reductions?

Reply:
Yep, reductions only apply to binary input unary output ufuncs